Chat qol changes

AnauNaga opened the pull request:
- Chat box can now be moved
- Chat box will now appear on the right side 
- Chat box location will save upon closing (Only in the current session)
- The chat icon now performs an animation

AnauNaga committed:
chat should blink but doesnt, box is now moveable

AnauNaga committed:
- The chat box can be moved
- The chat defaults to the right side of the screen
- The chat icon now performs a blinking animation (Todo: give users animation options)
- Added a intermediary texture for the chat icon

AnauNaga committed:
Small ChatBox qol changes

- The chat box can be moved
- The chat defaults to the right side of the screen
- The chat icon now performs a blinking animation (Todo: give users animation options)
- Added a intermediary texture for the chat icon

AnauNaga committed:
Merge branch 'ChatQolChanges' of https://github.com/RimworldTogether/Rimworld-Together into ChatQolChanges

AnauNaga committed:
Merge branch 'development' into ChatQolChanges

AnauNaga committed:
added mid chat icon files

AnauNaga committed:
Merge branch 'ChatQolChanges' of https://github.com/RimworldTogether/Rimworld-Together into ChatQolChanges

AnauNaga committed:
added ChatIconMid actually this time

Byte-Nova committed:
Revert "added mid chat icon files"

This reverts commit 71458a4b77b0bf6cd436b4977d7c8ef330daa764.

AnauNaga committed:
Fixed terrible chat Thread generation

AnauNaga committed:
Chat Dialog stays open

Byte-Nova commented:
Found the following issues while testing the PR:
- [ ] Current implementation of the chat thread continuously runs even if there's no reason for it, and stops the game's main thread each time an iteration is ran.

- [ ] Current implementation of the chat thread isn't correctly handled, and will sometimes cause the animation to run even if it shouldn't (Animation shouldn't play when the player has the chat open, and it still does sometimes because the thread checks using a timer).

- [ ] Chat index variable isn't reverted correctly to 0 each time the animation stops, which will cause jumps in the animation.

- [ ] Chat variables don't get a reset when connection is lost, meaning they will bleed into other sessions until the game closes.

- [ ] Chat notification sounds will play even if the chat is open, if this is a decision and not an error it should be configurable by the player to avoid being annoying.

- [ ] There are 2 chat tab files in the code that call to each other, if you wanted to implement a new chat tab do so using the same way the current tab uses (def injection from the mod folder).

- [ ] PR project contains post-build actions and assembly references that aren't shared within the project, this will make the CI fail and testing impossible for other developers.

Since the deadline for the new version is too close and we can't wait for the PR fixes I'll upload a different commit that contains all the implementations you are trying to push in this PR (excluding the latest one, making the chat always stay open), with fixes to the issues I found. If you are willing to implement more changes to the chat system you can always pull from the development branch and work from there when the 1.5 version is out.

Byte-Nova committed:
Commit for chat features (See description))

- Moved username variable to ClientValues.

- Implemented a safer thread execution for the chat manager.

- Implemented setting for the server to not display chat messages in the console.

- The chat tab now lets the game know when it's open.

- Chat sounds and animation won't play when the chat is open.

- Cleaned the ChatManager ReceiveMessages() function.

- Chat animation variables now reset correctly whenever the chat window is open.

- Chat variables now reset correctly whenever the user disconnects from the server.

//Overrides to previous PR

- Removed local assemblies and post-build events from the project manifest.

- Removed RT_Dialog_Chat.cs while it's not in use.

- Removed calling of the chat thread from the Network script and placed it in the ChatManager class.
